app/models/miq_worker.rb

```diff
@@ -345,9 +345,32 @@ def start_runner
     Process.detach(pid)
     pid
   end
+  alias_method :start_runner_via_fork, :start_runner
```

Comment:
The last commit that changes this and the code around is confusing and I don't understand why it is needed or how it works.  Seems like we should change the caller instead of jumping through hoops in the interface to make this work, but I admit that I still am unsure what you are fixing/trying to accomplish in that commit.